sample_app/lib/pages/channel_page.dart

```diff
@@ -124,12 +130,77 @@ class _ChannelPageState extends State<ChannelPage> {
             messageInputController: _messageInputController,
             onQuotedMessageCleared: _messageInputController.clearQuotedMessage,
             enableVoiceRecording: true,
+            allowedAttachmentPickerTypes: [
+              ...AttachmentPickerType.values,
+              if (config?.sharedLocations == true && channel.canShareLocation)
+                const LocationPickerType(),
+            ],
+            onCustomAttachmentPickerResult: (result) {
+              return _onCustomAttachmentPickerResult(channel, result).ignore();
+            },
+            customAttachmentPickerOptions: [
+              TabbedAttachmentPickerOption(
+                key: 'location-picker',
+                icon: const Icon(Icons.near_me_rounded),
+                supportedTypes: [const LocationPickerType()],
+                isEnabled: (value) {
+                  // Enable if nothing has been selected yet.
+                  if (value.isEmpty) return true;
+
+                  // Otherwise, enable only if there is a location.
+                  return value.extraData['location'] != null;
+                },
+                optionViewBuilder: (context, controller) => LocationPicker(
+                  onLocationPicked: (locationResult) {
+                    if (locationResult == null) return Navigator.pop(context);
+
+                    controller.extraData = {
+                      ...controller.value.extraData,
+                      'location': locationResult,
+                    };
+
+                    final result = LocationPicked(location: locationResult);
+                    return Navigator.pop(context, result);
+                  },
+                ),
+              ),
+            ],
           ),
         ],
       ),
     );
   }
 
+  Future<void> _onCustomAttachmentPickerResult(
+    Channel channel,
+    CustomAttachmentPickerResult result,
+  ) async {
+    final response = switch (result) {
+      LocationPicked() => _onShareLocationPicked(channel, result.location),
+      _ => null,
+    };
+
+    return response?.ignore();
+  }
+
+  Future<SendMessageResponse> _onShareLocationPicked(
+    Channel channel,
+    LocationPickerResult result,
+  ) async {
+    if (result.endSharingAt case final endSharingAt?) {
+      return channel.startLiveLocationSharing(
+        createdByDeviceId: 'test-device-id',
```

Comment:
Do we want to create a random UUID and store this locally in shared prefs or something, so we can identify devices?

Reply:
The `createdByDeviceId` is going to be made optional in backend soon, There's a PR yet to be merged. I will remove this once the backend is deployed.

Reply:
Removed